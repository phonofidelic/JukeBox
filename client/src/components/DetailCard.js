import React, { Component } from 'react';
import {
	Button,
	Dialog,
	DialogTitle,
	DialogContent,
	DialogContentText,
	DialogActions,
	Typography,	
} from '@material-ui/core';

const DUMMY_TEXT = `Lorem ipsum dolor sit amet, putent sadipscing id qui, vel ubique propriae contentiones ex, mel in nulla option. Eam ut mutat iuvaret, nihil saperet theophrastus sit ne, nulla zril euismod qui cu. Vel id indoctum conceptam rationibus. Prompta civibus eu eos. Purto ignota contentiones vel ne, numquam aliquid an vix, persius menandri delicatissimi mel an. Mea ne debet molestiae assentior, regione tibique antiopam no qui.
No vivendum voluptaria cum. Vis mucius intellegat ad, erat rationibus quo no, purto augue dolore ne quo. Scripta expetendis cum an, vero rebum interpretaris te eos. Wisi maiestatis his ad, sale scaevola salutatus in has. His ei numquam laboramus, te sea cetero persius, cum at alii adolescens.

Sed tollit dignissim an, cu eum affert tollit numquam, an mea feugait recusabo erroribus. Eam ei minim commodo, stet nullam ius no, quando oratio scribentur est ea. Ius doctus similique in, pri wisi nusquam signiferumque ea, docendi platonem no mea. In nec quidam deseruisse, vel ut erant posidonium. Has amet labores imperdiet id, fierent percipit tacimates an vel. Ius meis tibique at, inermis torquatos assentior in eam, sit denique propriae dissentiet an.

Quas oratio repudiandae cum ex, velit debet ad vel. Cu hinc audiam usu, cu usu tation veritus intellegam. Eu vix oporteat ocurreret, tollit legimus alienum nec an. Et quaeque admodum accusamus eos. Posse ancillae antiopam in sed, sit vero feugait signiferumque ei, mea debet laoreet et. Duo ullum vocibus definitiones et, veritus graecis minimum nec eu.

Ius feugait detraxit molestiae ut, in nam ocurreret neglegentur, ei placerat mediocritatem eam. Adhuc lobortis te qui, et ullum eleifend deseruisse usu. An vix eirmod sanctus epicurei, civibus ancillae torquatos nec at. Congue oratio facilis te quo, vim ea natum lorem offendit. Id magna gubergren adolescens sit, sed eu reque mentitum, case vidit mazim mel in.
Lorem ipsum dolor sit amet, putent sadipscing id qui, vel ubique propriae contentiones ex, mel in nulla option. Eam ut mutat iuvaret, nihil saperet theophrastus sit ne, nulla zril euismod qui cu. Vel id indoctum conceptam rationibus. Prompta civibus eu eos. Purto ignota contentiones vel ne, numquam aliquid an vix, persius menandri delicatissimi mel an. Mea ne debet molestiae assentior, regione tibique antiopam no qui.

No vivendum voluptaria cum. Vis mucius intellegat ad, erat rationibus quo no, purto augue dolore ne quo. Scripta expetendis cum an, vero rebum interpretaris te eos. Wisi maiestatis his ad, sale scaevola salutatus in has. His ei numquam laboramus, te sea cetero persius, cum at alii adolescens.

Sed tollit dignissim an, cu eum affert tollit numquam, an mea feugait recusabo erroribus. Eam ei minim commodo, stet nullam ius no, quando oratio scribentur est ea. Ius doctus similique in, pri wisi nusquam signiferumque ea, docendi platonem no mea. In nec quidam deseruisse, vel ut erant posidonium. Has amet labores imperdiet id, fierent percipit tacimates an vel. Ius meis tibique at, inermis torquatos assentior in eam, sit denique propriae dissentiet an.

Quas oratio repudiandae cum ex, velit debet ad vel. Cu hinc audiam usu, cu usu tation veritus intellegam. Eu vix oporteat ocurreret, tollit legimus alienum nec an. Et quaeque admodum accusamus eos. Posse ancillae antiopam in sed, sit vero feugait signiferumque ei, mea debet laoreet et. Duo ullum vocibus definitiones et, veritus graecis minimum nec eu.

Ius feugait detraxit molestiae ut, in nam ocurreret neglegentur, ei placerat mediocritatem eam. Adhuc lobortis te qui, et ullum eleifend deseruisse usu. An vix eirmod sanctus epicurei, civibus ancillae torquatos nec at. Congue oratio facilis te quo, vim ea natum lorem offendit. Id magna gubergren adolescens sit, sed eu reque mentitum, case vidit mazim mel in.
Lorem ipsum dolor sit amet, putent sadipscing id qui, vel ubique propriae contentiones ex, mel in nulla option. Eam ut mutat iuvaret, nihil saperet theophrastus sit ne, nulla zril euismod qui cu. Vel id indoctum conceptam rationibus. Prompta civibus eu eos. Purto ignota contentiones vel ne, numquam aliquid an vix, persius menandri delicatissimi mel an. Mea ne debet molestiae assentior, regione tibique antiopam no qui.

No vivendum voluptaria cum. Vis mucius intellegat ad, erat rationibus quo no, purto augue dolore ne quo. Scripta expetendis cum an, vero rebum interpretaris te eos. Wisi maiestatis his ad, sale scaevola salutatus in has. His ei numquam laboramus, te sea cetero persius, cum at alii adolescens.

Sed tollit dignissim an, cu eum affert tollit numquam, an mea feugait recusabo erroribus. Eam ei minim commodo, stet nullam ius no, quando oratio scribentur est ea. Ius doctus similique in, pri wisi nusquam signiferumque ea, docendi platonem no mea. In nec quidam deseruisse, vel ut erant posidonium. Has amet labores imperdiet id, fierent percipit tacimates an vel. Ius meis tibique at, inermis torquatos assentior in eam, sit denique propriae dissentiet an.

Quas oratio repudiandae cum ex, velit debet ad vel. Cu hinc audiam usu, cu usu tation veritus intellegam. Eu vix oporteat ocurreret, tollit legimus alienum nec an. Et quaeque admodum accusamus eos. Posse ancillae antiopam in sed, sit vero feugait signiferumque ei, mea debet laoreet et. Duo ullum vocibus definitiones et, veritus graecis minimum nec eu.

Ius feugait detraxit molestiae ut, in nam ocurreret neglegentur, ei placerat mediocritatem eam. Adhuc lobortis te qui, et ullum eleifend deseruisse usu. An vix eirmod sanctus epicurei, civibus ancillae torquatos nec at. Congue oratio facilis te quo, vim ea natum lorem offendit. Id magna gubergren adolescens sit, sed eu reque mentitum, case vidit mazim mel in.`;

class DetailCard extends Component {
	state = {
		// open: false
		isMobile: navigator.userAgent.indexOf('Mobile') > 0 ? true : false,
	}

	handleClose() {
		// this.setState({ open: false })
		this.props.handleCloseDetailView();
	}

	render() {
		const { detailViewData } = this.props;

		const { isMobile } = this.state;

		const imgWidth = isMobile ? window.innerWidth : 600;
		const imgHeight = isMobile ? window.innerWidth : 600;

		const PLACEHOLDER_IMAGE = `http://placekitten.com/${imgWidth}/${imgHeight}`;

		const styles = {
			root: {
				top: 0,
				// zIndex: 1,
			},
			cardMedia: {
				width: 600,
				height: 400,
				margin: 0,
			},

			dialog: {
				// width: '100%',
			},
			dialogContent: {
				padding: '0px',
			},
			dialogText: {
				paddingLeft: 24,
				paddingRight: 24,
			}
		}

		return (
			<div>
				<Dialog
					style={styles.dialog}
					fullScreen={isMobile}
					open={Boolean(detailViewData)}
					scroll="paper"
				>
					{	detailViewData &&
						<DialogContent style={styles.dialogContent}>
							<div>
								<img 
									src={detailViewData.imgSrc !== 'defaultImage' ? detailViewData.imgSrc : PLACEHOLDER_IMAGE} 
									width={imgWidth}
									height={imgHeight}
								/>
							</div>
							<DialogTitle>
								{ detailViewData.name || detailViewData.title }
							</DialogTitle>
							<DialogContentText style={styles.dialogText}>
								{ detailViewData.description }
							</DialogContentText>
						</DialogContent>
					}
					<DialogActions>
						<Button>
							More info
						</Button>
						<Button onClick={() => this.handleClose()}>
							Close
						</Button>
					</DialogActions>
				</Dialog>
			</div>
		);
	}
}

export default DetailCard;
