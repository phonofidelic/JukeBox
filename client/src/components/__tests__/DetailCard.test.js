import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import { createRender } from '@material-ui/core/test-utils';
import DetailCard from '../DetailCard';

const testArtistData = {
	name: 'Test',
	description: 'This is a test.'
};

const testAlbumData = {
	title: 'Test',
	description: 'This is a test.'
}

describe('DetailCard', () => {
	let props;
	let mountedDetailCard;
	let muiRenderer;

	const detailMessage = (props) => {
		if (!mountedDetailCard) {
			mountedDetailCard = mount(
				<DetailCard {...props} />
			);
		}
		return mountedDetailCard;
	};

	beforeEach(() => {
		props = {
			detailViewData: null,
			handleCloseDetailView: jest.fn()
		};
		mountedDetailCard = undefined;
		muiRenderer = createRender();
	});

	it('renders an empty div if no detailViewData is present', () => {
		const tree = renderer
			.create(<DetailCard {...props} />)
			.toJSON();
		const divs = detailMessage(props).find('div');

		expect(tree).toMatchSnapshot();
		expect(divs).toHaveLength(1);
	});

	it('renders a Dialog if detailViewData is present', () => {
		props.detailViewData = testArtistData;

		const tree = muiRenderer(<DetailCard {...props} />);
		const dialog = detailMessage(props).find('Dialog');

		expect(tree).toMatchSnapshot();
		expect(dialog).toHaveLength(1);
	});
});