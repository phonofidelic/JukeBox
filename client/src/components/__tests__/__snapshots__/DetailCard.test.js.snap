// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailCard renders a Dialog if detailViewData is present 1`] = `
initialize {
  "0": Object {
    "attribs": Object {},
    "children": Array [],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {},
    "x-attribsPrefix": Object {},
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`DetailCard renders an empty div if no detailViewData is present 1`] = `<div />`;
